cmd/cli/root.go

```diff
@@ -110,10 +85,12 @@ func NewRootCmd() *cobra.Command {
 			"BACALHAU_DIR or --repo must be set to initialize a node.\n\n", err)
 	}
 	RootCmd.PersistentFlags().String("repo", defaultRepo, "path to bacalhau repo")
-	if err := viper.BindPFlag("repo", RootCmd.PersistentFlags().Lookup("repo")); err != nil {
+
+	// Bind the repo flag to the system configuration
+	if err := cfg.System().BindPFlag("repo", RootCmd.PersistentFlags().Lookup("repo")); err != nil {
 		util.Fatal(RootCmd, err, 1)
 	}
-	if err := viper.BindEnv("repo", "BACALHAU_DIR"); err != nil {
+	if err := cfg.System().BindEnv("repo", "BACALHAU_DIR"); err != nil {
```

Comment:
The repo path is part of the 'System' config. And currently is the only value in that instance. Users are still free to specify a path for their repo.